### DISCUSSION TITLE
Completion suggestion does not pop up automatically for TypedDict

### DISCUSSION THREAD

**Discussion by kingyue737** [microsoft/pylance-release#3710] [Enhancement]
## Environment data

-   Language Server version: 2022.11.10 (pyright 4e50162d)
-   OS and version: Windows 11 21H2
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.7

## Code Snippet

```python
class Point(TypedDict):
    x: int
    y: int

p: Point = {""}
```

## Expected behavior

Two suggestions `x` and `y` should pop up automatically when user types the quotes.

## Actual behavior

No completion suggestion pops up. Need to manually trigger it by hitting `Ctrl Space`

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by Archmonger** [comment]
Agreed. This should be implemented.

**Comment by rmorshea** [comment]
I wasn't actually aware that you could manually request hints via `Ctrl+Space`. For quite a while I just thought this feature did not exist in Pylance. Only after finding this discussion did I realize Pylance could provide these suggestions. I expect that for the vast majority of users, if VSCode does not automatically make a hint, then they will assume that no hints exist.

**Reply by Archmonger** [reply]
Yep, I always assumed VSCode and/or Pylance didn't support `TypedDict` autocompletion.
